### ISSUE TITLE
Tensorflow Lite Custom op have no effect even source code *.cc has modified

### ISSUE THREAD

**Issue by pcub** [tensorflow/tensorflow#21643]
System information
== cat /etc/issue ===============================================
Linux master 4.15.0-29-generic #31~16.04.1-Ubuntu SMP Wed Jul 18 08:54:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
VERSION="16.04.5 LTS (Xenial Xerus)"
VERSION_ID="16.04"
VERSION_CODENAME=xenial

== are we in docker =============================================
No

== compiler =====================================================
c++ (Ubuntu 5.5.0-12ubuntu1~16.04) 5.5.0 20171010
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

== uname -a =====================================================
Linux master 4.15.0-29-generic #31~16.04.1-Ubuntu SMP Wed Jul 18 08:54:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

== check pips ===================================================
numpy 1.14.5
protobuf 3.6.0
tensorflow 1.8.0
tensorflow-tensorboard 0.4.0

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.8.0
tf.GIT_VERSION = v1.8.0-0-g93bc2e2072
tf.COMPILER_VERSION = v1.8.0-0-g93bc2e2072
Sanity check: array([1], dtype=int32)
/usr/local/lib/python2.7/dist-packages/h5py/init.py:36: FutureWarning: Conversion of the second argument of issubdtype from float to np.floating is deprecated. In future, it will be treated as np.float64 == np.dtype(float).type.
from ._conv import register_converters as _register_converters

== env ==========================================================
LD_LIBRARY_PATH /usr/local/cuda/lib64/:/usr/local/cuda/lib64/:/usr/local/cuda-9.0/lib64:
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
Mon Jul 30 11:49:23 2018
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.45 Driver Version: 396.45 |
|-------------------------------+----------------------+----------------------+
| GPU Name Persistence-M| Bus-Id Disp.A | Volatile Uncorr. ECC |
| Fan Temp Perf Pwr:Usage/Cap| Memory-Usage | GPU-Util Compute M. |
|===============================+======================+======================|
| 0 Quadro P600 Off | 00000000:03:00.0 On | N/A |
| 34% 43C P8 N/A / N/A | 574MiB / 1997MiB | 1% Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes: GPU Memory |
| GPU PID Type Process name Usage |
|=============================================================================|
| 0 1152 G /usr/lib/xorg/Xorg 237MiB |
| 0 2213 G compiz 79MiB |
| 0 2589 G ...-token=224D99F526E00AE3A3C0EF4D5E6D103A 113MiB |
| 0 5305 G ...-token=53BA9EE005E85645FEB6537828E37D64 141MiB |
+-----------------------------------------------------------------------------+

== cuda libs ===================================================
/usr/local/cuda-9.2/doc/man/man7/libcudart.7
/usr/local/cuda-9.2/doc/man/man7/libcudart.so.7
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart_static.a
/usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart.so.9.2.148

### Describe the problem
I converted caffe ssd model to tensorflow/tensorfow lite. There are small differences in caffe and tensorflow detection layer (i.e., decode box and coordinates difference). 
Tensorflow lite does not support all ops of tensorflow.
Detection layer in tensorflow lite is implemented in detect_postprocess.cc as custom op.
I changed detect_postprocess.cc to apply caffes' decode box and other functions. Then, I compiled tf with "bazel build -c opt --jobs=8 //tensorflow/tools/pip_package:build_pip_package" and also tried with "bazel build -c opt "//tensorflow/contrib/lite/kernels:builtin_ops",
However, problem is when I load detect.tflite on android and/or PC (with tensorfow lite python api <<tf.VERSION = 1.8.0 does not work,tf version:: 1.10.0 workd>>), the detection results return the same result as what detect_postprocess.cc supposed to work even I changed functionality of detect_postprocess.cc.

In other words, tensorflow lite always refers very first version of detect_postprocess.cc

What is the problem?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by aselle** [comment]
Why don't you just remove the registration of the default one. Or call your modified version a different name?

**Comment by achowdhery** [comment]
@drpngx Your op resolver is not able to resolve to the new code in the custom op. Can you share the steps that you ran after changing the code in detect_postprocess.cc

**Comment by pcub** [comment]
thank you, I will test it out.

**Comment by pcub** [comment]
@aselle the modified version with different names does not make any sense.
tflite c++ api with libtensorflow-lite.a (includes modified TFLite_Detection_PostProcess) works on pc.

**Comment by aselle** [comment]
You are not being clear on what you are doing.

You modified detect_postprocess.cc and recompiled right? Did you make a copy of it and add it to the build file, or did you modify it in place.

Does your modification work under any circumstance?

Did you compile a new AAR for android? or are you using the stock one? If the stock one then it of course will not work.

**Comment by pcub** [comment]
> You modified detect_postprocess.cc and recompiled right?
yes, I modified and recompiled.
I did not copy the detect_postprocess.cc to any place.
> Does your modification work under any circumstance?
Yes it does, I mentioned it "detect.tflite on c++ api with libtensorflow-lite.a (includes modified TFLite_Detection_PostProcess) works on pc"
> Did you compile a new AAR for android? 
I referred https://firebase.google.com/docs/ml-kit/android/use-custom-tflite and https://github.com/tensorflow/tensorflow/issues/18751.

**Comment by pcub** [comment]
figured out.

**Comment by aselle** [comment]
Glad you got it working. Could you post what the solution was so others can benefit from your work? Thanks.
